ambry-store/src/main/java/com/github/ambry/store/BlobStoreStats.java

```diff
@@ -1054,18 +1235,19 @@ private void processIndexSegmentEntriesBackward(IndexSegment indexSegment, Predi
           indexSegment.getFile().getName(), storeId);
 
       // valid index entries wrt log segment reference time
-      forEachValidIndexEntry(indexSegment, newScanResults.logSegmentForecastStartTimeMs, keyFinalStates, false,
-          entry -> {
+      forEachValidIndexEntry(indexSegment, newScanResults.logSegmentForecastStartTimeMsForDeleted,
+          newScanResults.logSegmentForecastStartTimeMsForExpired, keyFinalStates, false, entry -> {
```

Comment:
`removeFinalStateOnPut` will remove the final state of a storeKey when encountering PUT record, since PUT record is the first record for the storeKey. This is for reducing memory usage. We don't want to keep final states of all store key in the map throughout the iteration of all index segment. 

However, in this method, we will process same index segment twice, first time for calculating valid size of log segment, second time for calculating valid size for container. So when calculating for log segment, we don't remove the final state  since later we would need the final state when calculating for containers.